pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -328,41 +337,68 @@ type osqueryFilePaths struct {
 // In return, a structure of paths is returned that can be used to launch an
 // osqueryd instance. An error may be returned if the supplied parameters are
 // unacceptable.
-func calculateOsqueryPaths(rootDir, extensionSocketPath string) (*osqueryFilePaths, error) {
+func calculateOsqueryPaths(opts osqueryOptions) (*osqueryFilePaths, error) {
+
+	// Determine the path to the extension socket
+	extensionSocketPath := opts.extensionSocketPath
+	if extensionSocketPath == "" {
+		extensionSocketPath = socketPath(opts.rootDirectory)
+	}
+
+	extensionAutoloadPath := filepath.Join(opts.rootDirectory, "osquery.autoload")
+
+	osqueryFilePaths := &osqueryFilePaths{
+		pidfilePath:           filepath.Join(opts.rootDirectory, "osquery.pid"),
+		databasePath:          filepath.Join(opts.rootDirectory, "osquery.db"),
+		augeasPath:            filepath.Join(opts.rootDirectory, "augeas-lenses"),
+		extensionSocketPath:   extensionSocketPath,
+		extensionAutoloadPath: extensionAutoloadPath,
+		extensionPaths:        make([]string, len(opts.autoloadedExtensions)),
+	}
+
+	if len(opts.autoloadedExtensions) == 0 {
+
+		// delete the autoload file if there were no provided autoloaded extensions
```

Comment:
Good call! Though won't `os.Create` do that anyhow?

Reply:
good point, just I will just move os.Create up in the function so it's always called